tests/deepspeed/test_deepspeed.py

```diff
@@ -1051,3 +1076,38 @@ def test_lr_scheduler(self):
         ]
         with patch_environment(omp_num_threads=1):
             execute_subprocess_async(cmd, env=os.environ.copy())
+
+    @require_cuda
+    def test_basic_dynamo_run(self):
+        mod_file = inspect.getfile(accelerate.test_utils)
+        test_file_path = os.path.sep.join(
+            mod_file.split(os.path.sep)[:-1] + ["scripts", "external_deps", "test_performance.py"]
+        )
+        with tempfile.TemporaryDirectory() as dirpath:
+            cmd = [
+                "accelerate",
+                "launch",
+                "--num_processes=1",
+                "--num_machines=1",
+                "--machine_rank=0",
+                "--mixed_precision=fp16",
+                "--use_deepspeed",
+                "--gradient_accumulation_steps=1",
+                "--zero_stage=3",
+                "--offload_optimizer_device=none",
+                "--offload_param_device=none",
+                "--dynamo_backend=eager",
```

Comment:
The current test passed and deepspeed is using dynamo. However the `--dynamo_backend=eager` seems to be ignored by accelerate, I'm using the `ACCELERATE_DYNAMO_BACKEND` below for test. Do you have any clue about that? @pacman100

Reply:
Hello, I found the reason. Below code should be added in the function `prepare_deepspeed_cmd_env` at https://github.com/huggingface/accelerate/blob/910c1b6a8f78ef8a9120418d0d3a57fb9213f1ac/src/accelerate/utils/launch.py#L266:
```
try:
        dynamo_backend = DynamoBackend(args.dynamo_backend.upper())
    except ValueError:
        raise ValueError(
            f"Unknown dynamo backend: {args.dynamo_backend.upper()}. Choose between {DynamoBackend.list()}."
        )
    current_env["ACCELERATE_DYNAMO_BACKEND"] = dynamo_backend.value
    current_env["ACCELERATE_DYNAMO_MODE"] = args.dynamo_mode
    current_env["ACCELERATE_DYNAMO_USE_FULLGRAPH"] = str(args.dynamo_use_fullgraph)
    current_env["ACCELERATE_DYNAMO_USE_DYNAMIC"] = str(args.dynamo_use_dynamic)
```